presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergTypes.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg;
+
+import com.facebook.presto.common.type.TimestampType;
+import com.facebook.presto.common.type.TimestampWithTimeZoneType;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import static com.facebook.presto.hive.HiveCommonSessionProperties.PARQUET_BATCH_READ_OPTIMIZATION_ENABLED;
+import static com.facebook.presto.iceberg.IcebergQueryRunner.createIcebergQueryRunner;
+import static com.google.common.base.Preconditions.checkState;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestIcebergTypes
+        extends AbstractTestQueryFramework
+{
+    private QueryRunner batchReaderEnabledQueryRunner;
+
+    @Override
+    protected QueryRunner createQueryRunner() throws Exception
+    {
+        this.batchReaderEnabledQueryRunner = createIcebergQueryRunner(
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                ImmutableMap.of(PARQUET_BATCH_READ_OPTIMIZATION_ENABLED, "true"),
```

Comment:
I think there is no need to refactor the `IcebergQueryRunner.createIcebergQueryRunner(...)`, just set `hive.parquet-batch-read-optimization-enabled` to `true` in `extraProperties` would be ok.

Reply:
Maybe there is another way, but adding this to extra properties actually causes the tests to break since it is an unused property there. This sets it as a configuration property, but it needs to be a session property. The session is actually passed to the distributed query runner builder in it's constructor, so we need some way to add properties to that session before building the runner. This is why I decided to make changes to `IcebergQueryRunner`. Please let me know if this clears things up, if you have another approach in mind here I would certainly be open to it!

Reply:
Ooh, sorry for the overlooking that `hive.parquet-batch-read-optimization-enabled` belongs to the config in `presto-hive-common`. Since `presto-main` does not dependent on `presto-hive-common`, `TestingPrestoServer` does not inject `HiveCommonModule` either. So you are right currently we cannot set `hive.parquet-batch-read-optimization-enabled` in `extraProperties` here.